src/plugins/data/common/index_patterns/index_patterns/index_pattern.ts

```diff
@@ -519,8 +524,13 @@ export class IndexPattern implements IIndexPattern {
             // and ensure we ignore the key if the server response
             // is the same as the original response (since that is expected
             // if we made a change in that key)
-            const serverChangedKeys = Object.keys(updatedBody).filter((key) => {
-              return updatedBody[key] !== body[key] && this.originalBody[key] !== updatedBody[key];
+
+            const serverChangedKeys: string[] = [];
+            Object.entries(updatedBody).forEach(([key, value]) => {
+              // @ts-ignore
+              if (value !== body[key] && value !== this.originalBody[key]) {
```

Comment:
Better use casting:

```suggestion
              if (value !== (body as any)[key] && value !== this.originalBody[key]) {
```

Reply:
If you disable TypeScript, there is absolutely no error checking for that line:

![image](https://user-images.githubusercontent.com/9773803/91419249-7e78d880-e853-11ea-85ab-d3fbef7d1fbe.png)

If you cast, you still see all the errors:

![image](https://user-images.githubusercontent.com/9773803/91419309-918ba880-e853-11ea-8713-e30cee798447.png)